cmd/minikube/cmd/service.go

```diff
@@ -67,3 +95,43 @@ func init() {
 	serviceCmd.Flags().BoolVar(&serviceURLMode, "url", false, "Display the kubernetes service URL in the CLI instead of opening it in the default browser")
 	RootCmd.AddCommand(serviceCmd)
 }
+
+func CheckService(namespace string, service string) func() error {
+	return func() error {
+		//get service
+		services, err := cluster.GetKubernetesServicesWithNamespace(namespace)
+		svc, err := services.Get(service)
+
+		//create a selector to get the pods the service points to
+		selector := labels.NewSelector()
+		for key, val := range svc.Spec.Selector {
+			requirement, err := labels.NewRequirement(key, "=", sets.NewString(val))
+			if err != nil {
+				return err
+			}
+			selector.Add(*requirement)
+		}
+
+		//get the pods the service points to
+		pods, err := cluster.GetKubernetesPodsWithNamespace(namespace)
```

Comment:
I don't think I can use endpoints as I don't believe there is a status on endpoints.  The endpoint seems to be up and operational immediately although the pods it points to are not ready to serve.  I think I will keep the original implementation as I don' understand how to "wait until ready" using the service or endpoint objects.  Please let me know if I am misunderstanding something, thanks!